### PR TITLE
Kubeadm remove v1alpha2 api

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR removes v1alpha2 API from kubeadm.

**Which issue(s) this PR fixes**:
Ref: https://github.com/kubernetes/kubeadm/issues/911

**Special notes for your reviewer**:
This PR is the first of a series or PRs that will remove v1alpha2, cleanup all the references to v1alpha2 objects, and finally add v1beta1 api version; the work is divided into multiple PRs hopefully more easy to review.

This PR leaves some TODOs as a placeholder for implementing v1alpha3-->v1beta1 roundtrip tests in following PRs. 

**Release note**:
```release-note
[action required] kubeadm: The `v1alpha2` config API has been removed.
Please convert your `v1alpha2` configuration files to `v1alpha3` using the
`kubeadm config migrate` command of kubeadm v1.12.x
```

@kubernetes/sig-cluster-lifecycle-pr-reviews
sig cluster-lifecycle
kind api-change
area kubeadm

/cc @timothysc @luxas

### CODE REVIEW

**Comment by neolit123** [comment]
thanks @fabriziopandini 
i will have a look a bit later.

/kind api-change

**Comment by timothysc** [comment]
Please merge this with https://github.com/kubernetes/kubernetes/pull/69056 into a single changeset for v1alpha2 for historical purposes.

**Comment by timothysc** [comment]
Also can you squash commits to 2.  

- removal of v1alpha2 
- autogenerated

**Comment by fabriziopandini** [comment]
Dear k8s ci robot
/test pull-kubernetes-e2e-kops-aws
/test pull-kubernetes-e2e-gce-100-performance

File: cmd/kubeadm/app/util/config/common.go [hunk 1]
```diff
@@ -79,11 +79,11 @@ func DetectUnsupportedVersion(b []byte) error {
...
 	// v1.10 and earlier: v1alpha1
 	// v1.11: v1alpha1 read-only, writes only v1alpha2 config
-	// v1.12: v1alpha2 read-only, writes only v1beta1 config. Warns if the user tries to use v1alpha1
-	// v1.13 and v1.14: v1beta1 read-only, writes only v1 config. Warns if the user tries to use v1alpha1 or v1alpha2.
-	// v1.15: v1 is the only supported format.
+	// v1.12: v1alpha2 read-only, writes only v1alpha3 config. Warns if the user tries to use v1alpha1
+	// v1.13: v1alpha3 read-only, writes only v1beta1 config. Warns if the user tries to use v1alpha1 or v1alpha2
 	oldKnownAPIVersions := map[string]string{
 		"kubeadm.k8s.io/v1alpha1": "v1.11",
+		"kubeadm.k8s.io/v1alpha2": "v1.12",
```
Comment (timothysc) [inline cmd/kubeadm/app/util/config/common.go]: b/c you are removing alpha 2 should you remove v1.11?

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

Please address the comment, or possibly follow up on a separate PR.

**Comment by chuckha** [comment]
Is the goal here to have one alpha version and n beta versions?

Seems like removing the v1alpha2 type is just as dangerous as making a breaking change to v1alpha1 like in the 1.10 release. It feels like at this point we'd want to rename v1alpha2 to v1alpha1 and v1alpha3 to v1beta1.

**Comment by timothysc** [comment]
@chuckha - It's explicitly stated that the support model for kubeadm needs to upgrade through.  So a 1.11 version can only upgrade from 1.10.  Maintaining code and machinery of past versions (that aren't supported) is an albatross so we continually shed the code every cycle.

**Comment by chuckha** [comment]
@timothysc right, I get that, but if I'm reading this right, we're getting rid of the newer version (v1alpha2) and keeping the older one (v1alpha1), I'd suggest renaming v1alpha2 to v1alpha1 and getting rid of the current v1alpha1.

File: cmd/kubeadm/app/util/config/common.go (hunk 1 above)
Comment (fabriziopandini) [reply cmd/kubeadm/app/util/config/common.go]: This piece of code gives warning to user if the pass to kubeadm an older version of the config, and tell what version of kubeadm can be used for executing config migrate.
I think that giving a warning also for v1alpha1 doesn't hurts...

**Comment by fabriziopandini** [comment]
@chuckha 
>  I get that, but if I'm reading this right, we're getting rid of the newer version (v1alpha2) and keeping the older one (v1alpha1)

Current version is v1alpha3; this PR is getting rid of v1alpha2
Soonish I will add v1beta1 and all the machinery for converting v1alpha3->v1beta1

@neolit123 
> i will send a PR for k-a, because i'm pretty sire this PR will make the master tests red.

Thanks. We can make k/a use v1alpha3 for kubeadm v>=1.12

**Pull request merged**